include/Contact.php

```diff
@@ -662,4 +664,34 @@ function posts_from_contact($a, $contact_id) {
 
 	return $o;
 }
+
+/**
+ * @brief Returns a formatted location string from the given profile array
+ *
+ * @param array $profile Profile array (Generated from the "profile" table)
+ *
+ * @return string Location string
+ */
+function formatted_location($profile) {
```

Comment:
You don't have time in the next days? I'm having holidays. We could do a "mini hackathon" where we could discuss stuff like this.